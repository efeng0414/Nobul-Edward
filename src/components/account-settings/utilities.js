import React from "react";
import SettingsFieldItem from "../settings/common/settings-field-item";
import {
  FIRST_NAME,
  LAST_NAME,
  PHONE,
  ADDRESS1,
  ADDRESS2,
  COUNTRY,
  PROVINCE_OR_STATE,
  CITY,
  POSTAL_OR_ZIP_CODE,
  PASSWORD,
  EMAIL
} from "../../../core/api-transform/users";
import TextField from "../form-fields/text-field";
import PhoneField from "../form-fields/phone-field";
import PostalField from "../form-fields/postal-field";
import CountryField from "../form-fields/country-field";
import ProvinceField from "../form-fields/province-field";
import ChangePasswordForm from "../forms/common/change-password-form";
import ChangeEmailForm from "../forms/common/change-email-form";

const LOCALE_KEY_PREFIX = "consumerSettings.account.";

export const fieldsMap = {
  [FIRST_NAME]: {
    labelKey: "firstName",
    type: TextField,
    required: true
  },
  [LAST_NAME]: {
    labelKey: "lastName",
    type: TextField,
    required: true
  },
  [EMAIL]: {
    labelKey: "email",
    type: ChangeEmailForm,
    required: true
  },
  [PASSWORD]: {
    labelKey: "password",
    type: ChangePasswordForm,
    required: true
  },
  [PHONE]: {
    labelKey: "phoneNumber",
    type: PhoneField,
    required: true
  },
  [ADDRESS1]: {
    labelKey: "address1",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [ADDRESS2]: {
    labelKey: "address2",
    type: TextField,
    halfWidth: true,
    required: false
  },
  [COUNTRY]: {
    labelKey: "country",
    type: CountryField,
    halfWidth: true,
    required: true
  },
  [PROVINCE_OR_STATE]: {
    labelKey: "province",
    type: ProvinceField,
    halfWidth: true,
    required: true
  },
  [CITY]: {
    labelKey: "city",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [POSTAL_OR_ZIP_CODE]: {
    labelKey: "postcode",
    type: PostalField,
    halfWidth: true,
    required: true
  }
};

export const generateSettingFieldsItemForProfile = ({
  userProfile,
  onEditClick
}) => {
  return Object.entries(fieldsMap).map(([key, { labelKey, halfWidth }]) => {
    return (
      <SettingsFieldItem
        labelKey={`${LOCALE_KEY_PREFIX}${labelKey}`}
        value={userProfile[key]}
        onEditClick={onEditClick}
        key={key}
        id={key}
        halfWidth={halfWidth}
      />
    );
  });
};
