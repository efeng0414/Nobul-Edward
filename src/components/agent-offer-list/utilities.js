import { REBATE_COMMISSION } from "../../../core/api-transform/users";
import { REBATE_COMMISSION as OFFER_REBATE_COMMISSION } from "../../../core/api-transform/offers";
import { BUY, SELL } from "../../../core/api-transform/jobs";

export const getJobType = ({ jobDetails, jobTypeProp }) => {
  if (jobTypeProp) return jobTypeProp;
  return jobDetails[REBATE_COMMISSION] !== undefined ||
    jobDetails[OFFER_REBATE_COMMISSION] !== undefined
    ? BUY
    : SELL;
};
