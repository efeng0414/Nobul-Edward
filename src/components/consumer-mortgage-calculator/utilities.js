export const amortizationRange = [5, 10, 15, 20, 25, 30, 35];
