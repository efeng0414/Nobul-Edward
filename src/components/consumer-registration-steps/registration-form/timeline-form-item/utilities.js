export const optionArray = [
  { value: 1, key: "timeline1" },
  { value: 2, key: "timeline2" },
  { value: 3, key: "timeline3" },
  { value: 4, key: "timeline4" },
  { value: 5, key: "timeline5" }
];
