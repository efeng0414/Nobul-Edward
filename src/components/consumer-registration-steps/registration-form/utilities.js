import { BUY, SELL, BOTH } from "../../../../core/constants/shared";

export const fieldMap = {
  LOOKING_FOR: "lookingFor",
  TIMELINE: "timeline",
  PRE_APPROVED_MORTGAGE: "preApprovedMortgage",
  AREA_OF_INTEREST: "areaOfInterest",
  TERM: "term",
  CONSENT: "consent",
  MORTGAGE_AMOUNT: "mortgageAmount",
  CONTRACT_TO_BROKERAGE: "contractToBrokerage",
  CONSENT_FOR_NEWS: "consent_for_news",
  CONSENT_FOR_EMAIL: "email_notifications"
};

export const getFormPayload = ({ signUpValues, formValues }) => {
  const postData = {
    ...signUpValues,
    ...formValues
  };
  postData.mortgageAmount = !postData.preApprovedMortgage
    ? "0"
    : postData.mortgageAmount.replace(/^0+/, "");

  postData.consentForNews = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_NEWS
  );

  postData.allowEmailNotifications = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_EMAIL
  );

  postData[fieldMap.LOOKING_FOR] = postData[fieldMap.LOOKING_FOR].includes(BOTH)
    ? [BUY, SELL]
    : [postData[fieldMap.LOOKING_FOR]];

  return postData;
};
