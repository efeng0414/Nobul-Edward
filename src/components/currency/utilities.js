export const CURRENCY_SYMBOLS = {
  CA: "<span class='currency__code'>CA</span>$",
  US: "<span class='currency__code'>US</span>$",
  EU: "€",
  GB: "£"
};
