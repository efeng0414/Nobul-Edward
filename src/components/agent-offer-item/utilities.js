import React from "react";
import { Icon, Tag } from "antd";
import { JOB_ACCEPTED, JOB_DELETED } from "../../../core/constants/jobs";
import {
  OFFER_OPEN,
  OFFER_ACCEPTED,
  OFFER_REJECTED,
  OFFER_WITHDRAWN,
  OFFER_PENDING_VERIFICATION
} from "../../../core/constants/offers";
import { translate } from "../../utilities/locale";

export const TAG_MAP = {
  ACCEPTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw &&
      correspondingJob.status === JOB_ACCEPTED &&
      offerDetail.status === OFFER_ACCEPTED,
    jsx: ({ intl }) => (
      <Icon type="check" key="ACCEPTED">
        {translate(intl, "offerAccepted")}
      </Icon>
    )
  },
  OPEN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_OPEN,
    jsx: ({ intl }) => (
      <Tag className="agent-offer__tag agent-offer__tag--open" key="OPEN">
        {translate(intl, "offerOpen")}
      </Tag>
    )
  },
  REJECTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_REJECTED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--rejected"
        key="REJECTED"
      >
        {translate(intl, "offerRejected")}
      </Tag>
    )
  },
  WITHDRAWN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      withdraw || offerDetail.status === OFFER_WITHDRAWN,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--withdrawn"
        key="WITHDRAWN"
      >
        {translate(intl, "offerWithdrawn")}
      </Tag>
    )
  },

  JOB_DELETED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && correspondingJob.status === JOB_DELETED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--job-deleted"
        key="JOB_DELETED"
      >
        {translate(intl, "offer.jobDeletedTag")}
      </Tag>
    )
  },

  PENDING_VERIFICATION: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_PENDING_VERIFICATION,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--pending-verification"
        key="OPEN"
      >
        {translate(intl, "offerPendingVerification")}
      </Tag>
    )
  }
};
