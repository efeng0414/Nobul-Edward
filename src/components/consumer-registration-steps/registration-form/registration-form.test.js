import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import RegistrationFormWrapped from "./index";
import { CONSUMER_SIGN_UP_SUBMIT } from "../../../utilities/google-tag-variable";
import { CONSUMER_USER_TYPE } from "../../../../core/constants/users";
import { AUTH_WRONG_PASSWORD } from "../../../../core/constants/firebaseErrorCode";

const {
  WrappedComponent: { WrappedComponent: RegistrationForm }
} = RegistrationFormWrapped;

describe("Registration Form", () => {
  const componentProps = {
    intl: global.getIntl(),
    goBackToHomePage: () => {},
    previousPage: () => {},
    signUpUser: () => {},
    nextPage: () => {},
    error: {},
    currentUser: {},
    hasContractToBrokerage: () => {},
    form: {
      getFieldDecorator: config => jsx => jsx,
      setFieldsValue: function(values) {
        this.setFieldValueCalledWith = values;
      },
      validateFieldsAndScroll: callback => {
        callback();
      },
      setFieldValueCalledWith: {}
    },
    signUpValues: {}
  };

  const mockEvent = {
    preventDefault: () => {}
  };

  const componentJSX = <RegistrationForm {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.getIntl()}>{componentJSX}</IntlProvider>
    );

    it("should match stored snapshot", () => {
      const componentSnapshot = renderer.create(snapshotJSX).toJSON();

      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component Render", () => {
    const functionsToReset = ["handleFormValidationCallback", "handleSignUp"];
    const componentPrototypeResets = functionsToReset.reduce(
      global.createComponentPrototypeResets({ component: RegistrationForm }),
      {}
    );
    const componentMount = Enzyme.mount(
      componentJSX,
      global.getIntlContext({ intl: componentProps.intl })
    );

    beforeEach(() => {
      functionsToReset.map(
        global.resetPropertyComponentProperty({
          component: RegistrationForm,
          componentMount,
          componentPrototypeResets
        })
      );
    });

    describe("ComponentDidMount", () => {
      it("should call form.setFieldsValue with the signUpValues or state values", () => {
        componentProps.signUpValues = {
          lookingFor: "bloop",
          areaOfInterest: "hey",
          consent: []
        };

        const componentMount = Enzyme.mount(
          <RegistrationForm {...componentProps} />,
          global.getIntlContext({ intl: componentProps.intl })
        );

        const expectedObject = {
          ...componentProps.signUpValues,
          timeline: componentMount.state().timeline,
          contractToBrokerage: componentMount.state().contractToBrokerage
        };

        expect(componentProps.form.setFieldValueCalledWith).toMatchObject(
          expectedObject
        );
      });
    });

    describe("inputMortgage function", () => {
      it("should set the allowInputMortage state value to that of the event.target.value", () => {
        componentMount.setState({ allowInputMortgage: false });
        const mockEvent = {
          target: {
            value: true
          }
        };

        componentMount.instance().inputMortgage(mockEvent);

        expect(componentMount.state().allowInputMortgage).toBe(
          mockEvent.target.value
        );
      });
    });

    describe("validateTerm function", () => {
      it("should callback with an error message if value does not include accept_term", () => {
        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        const mockValue = [];

        componentMount.instance().validateTerm(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBeTruthy();
      });

      it("should not callback with an error message if value does include accept_term ", () => {
        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        const mockValue = ["accept_term"];

        componentMount.instance().validateTerm(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBe(undefined);
      });
    });

    describe("handleSignUp function", () => {
      beforeEach(() => {
        componentMount.setProps({ getFormPayload: undefined });
      });

      it("should call preventDefault on the event", () => {
        const mockEvent = {
          preventDefault: function() {
            this.preventDefaultCalled = true;
          },
          preventDefaultCalled: false
        };

        componentMount.setProps({ getFormPayload: () => ({}) });

        componentMount.instance().handleSignUp(mockEvent);

        expect(mockEvent.preventDefaultCalled).toBe(true);
      });

      it("should call validateFieldsAndScroll", () => {
        const watchObject = {
          validateFieldsAndScroll: false
        };

        const mockValidateAndScroll = () => {
          watchObject.validateFieldsAndScroll = true;
        };
        componentProps.form.validateFieldsAndScroll = mockValidateAndScroll;
        componentMount.setProps(componentProps);
        componentMount.instance().handleSignUp(mockEvent);

        expect(watchObject.validateFieldsAndScroll).toBe(true);
      });

      it("should call handleFormValidationCallback", () => {
        const mockHandleFormValidationCallback = jest.fn();
        componentMount.instance().handleFormValidationCallback = mockHandleFormValidationCallback;

        const mockValidateAndScroll = callback => {
          callback();
        };
        componentProps.form.validateFieldsAndScroll = mockValidateAndScroll;
        componentMount.setProps(componentProps);
        componentMount.instance().handleSignUp(mockEvent);

        expect(mockHandleFormValidationCallback).toHaveBeenCalled();
      });
    });

    describe("handleFormValidation function", () => {
      it("should not update isSubmitted state if there are errors", () => {
        componentMount.setState({ hasSubmitted: false });
        componentMount
          .instance()
          .handleFormValidationCallback({ message: "error" }, { value: "a " });

        expect(componentMount.state().hasSubmitted).toBe(false);
      });

      it("should update isSubmitted state if there is no error", () => {
        componentMount.setState({ hasSubmitted: false });
        componentMount
          .instance()
          .handleFormValidationCallback(false, { value: "a " });

        expect(componentMount.state().hasSubmitted).toBe(true);
      });

      it("should call hasContractToBrokerage and nextPage if contractToBrokerage is true", () => {
        const hasContractToBrokerageMock = jest.fn();
        componentProps.hasContractToBrokerage = hasContractToBrokerageMock;

        const nextPageMock = jest.fn();
        componentProps.nextPage = nextPageMock;

        componentMount.setProps(componentProps);
        componentMount
          .instance()
          .handleFormValidationCallback(false, { contractToBrokerage: true });

        expect(hasContractToBrokerageMock).toHaveBeenCalled();
        expect(nextPageMock).toHaveBeenCalled();
      });

      it("should call gtmEvent with correct event if no contractToBrokerage", () => {
        componentProps.gtmEvent = jest.fn();
        componentMount.setProps(componentProps);

        componentMount
          .instance()
          .handleFormValidationCallback(false, { contractToBrokerage: false });

        const expectedObject = {
          name: CONSUMER_SIGN_UP_SUBMIT
        };
        expect(componentProps.gtmEvent).toHaveBeenCalledWith(
          expect.objectContaining(expectedObject)
        );
      });

      it("should call signUp user with the post data and consumer user type", () => {
        componentProps.signUpUser = jest.fn();
        componentProps.signUpValues = {
          a: 1,
          timeline: 1,
          contractToBrokerage: false,
          areaOfInterest: "hey",
          lookingFor: "both"
        };

        componentMount.setProps(componentProps);

        const values = {
          contractToBrokerage: false,
          consent: [],
          preApprovedMortgage: false
        };
        componentMount.instance().handleFormValidationCallback(false, values);

        const expectedObject = {
          postData: {
            ...values,
            ...componentProps.signUpValues,
            allowEmailNotifications: false,
            consentForNews: false,
            mortgageAmount: "0",
            lookingFor: ["buy", "sell"]
          },
          userType: CONSUMER_USER_TYPE
        };
        expect(
          componentProps.signUpUser.mock.calls[0][0].postData
        ).toMatchObject(expectedObject.postData);
        expect(componentProps.signUpUser.mock.calls[0][0].userType).toEqual(
          expectedObject.userType
        );
      });
    });

    describe("ComponentDidUpdate function", () => {
      it("should call props.goBackToHomePage if user is emailVerified", () => {
        componentProps.currentUser = {
          emailVerified: true
        };
        componentProps.goBackToHomePage = jest.fn();

        componentMount.setProps(componentProps);

        expect(componentProps.goBackToHomePage).toHaveBeenCalled();
      });

      it("should not call goBackToHomePage if user is not emailVerified", () => {
        componentProps.currentUser = {
          emailVerified: false
        };
        componentProps.goBackToHomePage = jest.fn();

        componentMount.setProps(componentProps);

        expect(componentProps.goBackToHomePage).not.toHaveBeenCalled();
      });

      it("should call message.info if error introduced in update", () => {
        componentProps.error = {};
        componentMount.setProps(componentProps);
        componentProps.error = { code: AUTH_WRONG_PASSWORD };
        componentProps.message = {
          info: jest.fn()
        };
        componentMount.setProps(componentProps);
        expect(componentProps.message.info).toHaveBeenCalled();
      });
    });

    describe("render function", () => {
      describe("Form", () => {
        it("should call handleSignUp function on form submit", () => {
          const handleSignUpMock = jest.fn();
          componentMount.instance().handleSignUp = handleSignUpMock;
          componentMount.setProps({ a: 1 });

          componentMount
            .find(".consumer-registration-form-form")
            .at(0)
            .simulate("submit");

          expect(handleSignUpMock).toHaveBeenCalled();
        });
      });
    });
  });
});
