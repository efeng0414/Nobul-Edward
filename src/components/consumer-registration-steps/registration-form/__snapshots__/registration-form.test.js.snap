// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Component Snapshot should match stored snapshot 1`] = `
<div
  className="consumer-registration"
>
  <div
    className="consumer-registration-img consumer-registration-img-2"
  >
    <img
      className="consumer-registration-logo"
      src={
        Object {
          "default": "file-mock",
        }
      }
    />
  </div>
  <div
    className="consumer-registration-form registration-container"
  >
    <h2
      className="header-step header-step-2"
    >
      Join Nobul
    </h2>
    <form
      className="ant-form ant-form-horizontal consumer-registration-form-form"
      onSubmit={[Function]}
    >
      <div>
        <div
          className="ant-row"
        >
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="First Name"
                >
                  First Name
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      onKeyDown={[Function]}
                      type="text"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="Last Name"
                >
                  Last Name
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      onKeyDown={[Function]}
                      type="text"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row"
        >
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="Email"
                >
                  Email
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      onKeyDown={[Function]}
                      type="text"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="Phone Number"
                >
                  Phone Number
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <span
                      className="ant-input-group-wrapper"
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    >
                      <span
                        className="ant-input-wrapper ant-input-group"
                      >
                        <span
                          className="ant-input-group-addon"
                        >
                          <div
                            className="ant-select ant-select-enabled"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            style={
                              Object {
                                "width": 70,
                              }
                            }
                          >
                            <div
                              aria-autocomplete="list"
                              aria-expanded={false}
                              aria-haspopup="true"
                              className="ant-select-selection
            ant-select-selection--single"
                              onKeyDown={[Function]}
                              role="combobox"
                              tabIndex={0}
                            >
                              <div
                                className="ant-select-selection__rendered"
                              />
                              <span
                                className="ant-select-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "WebkitUserSelect": "none",
                                    "userSelect": "none",
                                  }
                                }
                                unselectable="on"
                              >
                                <b />
                              </span>
                            </div>
                          </div>
                        </span>
                        <input
                          className="ant-input"
                          disabled={false}
                          onKeyDown={[Function]}
                          style={null}
                          type="text"
                        />
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row"
        >
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="Password"
                >
                  Password
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <span
                      className="ant-input-affix-wrapper"
                    >
                      <input
                        className="ant-input"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        style={null}
                        type="password"
                      />
                      <span
                        className="ant-input-suffix"
                      >
                        <i
                          className="anticon anticon-eye"
                          onClick={[Function]}
                        />
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-col-xs-12 ant-col-sm-6 ant-col-lg-6"
            mg={6}
          >
            <div
              className="ant-row ant-form-item"
            >
              <div
                className="ant-form-item-label"
              >
                <label
                  className=""
                  htmlFor={null}
                  onClick={[Function]}
                  title="Confirm Password"
                >
                  Confirm Password
                </label>
              </div>
              <div
                className="ant-form-item-control-wrapper"
              >
                <div
                  className="ant-form-item-control"
                >
                  <span
                    className="ant-form-item-children"
                  >
                    <span
                      className="ant-input-affix-wrapper"
                    >
                      <input
                        className="ant-input"
                        disabled={false}
                        onKeyDown={[Function]}
                        style={null}
                        type="password"
                      />
                      <span
                        className="ant-input-suffix"
                      >
                        <i
                          className="anticon anticon-eye"
                          onClick={[Function]}
                        />
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="consumer-registration-form-item contract-brokerage"
        >
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <p
                    className="consumer-registration-paragraph"
                  >
                    Are you currently under contract to a brokerage?
                  </p>
                  <div
                    className="ant-radio-group ant-radio-group-outline"
                  >
                    <label
                      className="ant-radio-wrapper"
                    >
                      <span
                        className="ant-radio"
                        style={Object {}}
                      >
                        <input
                          checked={false}
                          className="ant-radio-input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="radio"
                          value={false}
                        />
                        <span
                          className="ant-radio-inner"
                        />
                      </span>
                      <span>
                        No
                      </span>
                    </label>
                    <label
                      className="ant-radio-wrapper"
                    >
                      <span
                        className="ant-radio"
                        style={Object {}}
                      >
                        <input
                          checked={false}
                          className="ant-radio-input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="radio"
                          value={true}
                        />
                        <span
                          className="ant-radio-inner"
                        />
                      </span>
                      <span>
                        Yes
                      </span>
                    </label>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="consumer-registration-form-item"
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <p
                  className="consumer-registration-paragraph"
                >
                  Are you looking to buy or sell?
                </p>
                <div
                  className="ant-radio-group ant-radio-group-outline"
                >
                  <label
                    className="ant-radio-wrapper"
                  >
                    <span
                      className="ant-radio"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-radio-input"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="radio"
                        value="both"
                      />
                      <span
                        className="ant-radio-inner"
                      />
                    </span>
                    <span>
                      Both
                    </span>
                  </label>
                  <label
                    className="ant-radio-wrapper"
                  >
                    <span
                      className="ant-radio"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-radio-input"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="radio"
                        value="buy"
                      />
                      <span
                        className="ant-radio-inner"
                      />
                    </span>
                    <span>
                      Buy
                    </span>
                  </label>
                  <label
                    className="ant-radio-wrapper"
                  >
                    <span
                      className="ant-radio"
                      style={Object {}}
                    >
                      <input
                        checked={false}
                        className="ant-radio-input"
                        disabled={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="radio"
                        value="sell"
                      />
                      <span
                        className="ant-radio-inner"
                      />
                    </span>
                    <span>
                      Sell
                    </span>
                  </label>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="consumer-registration-form-item"
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <p
                  className="consumer-registration-paragraph"
                >
                  When in the near future are you looking buy or sell?
                </p>
                <div
                  className="consumer-registration-select ant-select ant-select-enabled"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup="true"
                    className="ant-select-selection
            ant-select-selection--single"
                    onKeyDown={[Function]}
                    role="combobox"
                    tabIndex={0}
                  >
                    <div
                      className="ant-select-selection__rendered"
                    />
                    <span
                      className="ant-select-arrow"
                      onClick={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <b />
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="consumer-registration-form-item consumer-registration-form-checkboxes"
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <div
                  className="ant-checkbox-group"
                >
                  <div
                    className="consumer-registration-accept-terms-group"
                  >
                    <label
                      className="ant-checkbox-wrapper"
                    >
                      <span
                        className="ant-checkbox"
                        style={Object {}}
                      >
                        <input
                          checked={false}
                          className="ant-checkbox-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="checkbox"
                          value="accept_term"
                        />
                        <span
                          className="ant-checkbox-inner"
                        />
                      </span>
                      <span>
                        <div>
                          By checking this box you confirm that you have read and agree to the 
                          <span
                            className="accept-terms-link"
                          >
                            <a
                              href="https://corp.nobul.com/legal/terms-and-conditions/"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              Terms of Service
                            </a>
                          </span>
                           and that you have read our Privacy Policy.
                        </div>
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  className="ant-checkbox-group"
                >
                  <div
                    className="consumer-registration-accept-terms-group"
                  >
                    <label
                      className="ant-checkbox-wrapper"
                    >
                      <span
                        className="ant-checkbox"
                        style={Object {}}
                      >
                        <input
                          checked={false}
                          className="ant-checkbox-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="checkbox"
                          value="consent_for_news"
                        />
                        <span
                          className="ant-checkbox-inner"
                        />
                      </span>
                      <span>
                        <div>
                          I consent to receiving news and updates about nobul’s products and services. More information can be found in our 
                          <span
                            className="accept-terms-link"
                          >
                            <a
                              href="https://corp.nobul.com/legal/privacy-policy/"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              Privacy Policy
                            </a>
                          </span>
                           or 
                          <span
                            className="accept-terms-link"
                          >
                            <a
                              href="https://corp.nobul.com/contact-us/"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              Contact Us
                            </a>
                          </span>
                          . You may withdraw consent or unsubscribe at any time.
                        </div>
                      </span>
                    </label>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="consumer-registration-button-group"
      >
        <div
          className="ant-row ant-form-item"
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <button
                  className="ant-btn consumer-registration-button-item button-next ant-btn-primary ant-btn-lg"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Sign Up
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
